tensorboard/schema.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard database schema module.
+
+Currently the only database supported is SQLite.
+
+Warning: This module is EXPERIMENTAL. It will not be considered stable
+until data migration tools are put into place. Until that time, any
+database created with this schema will need to be deleted as new
+updates occur.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import contextlib
+
+
+def setup_database(db_conn):
+  """Creates SQL tables and indexes needed by TensorBoard.
+
+  If the tables and indexes are already created, this function has no
+  effect. Commit is called on the connection object when this function
+  is complete.
+
+  Args:
+    db_conn: A PEP249 Connection object.
+  """
+  create_runs_table(db_conn)
+  create_event_logs_table(db_conn)
+  db_conn.commit()
+
+
+def create_runs_table(db_conn):
+  """Creates the runs table."""
+  with contextlib.closing(db_conn.cursor()) as c:
+    c.execute('''\
+      CREATE TABLE IF NOT EXISTS runs (
+        id VARCHAR(64) PRIMARY KEY,
+        name VARCHAR(255) NOT NULL
+      )
+    ''')
+    c.execute('''\
+      CREATE UNIQUE INDEX IF NOT EXISTS runs_index
+      ON runs (name)
+    ''')
+
+
+def create_event_logs_table(db_conn):
```

Comment:
What is an "event log"? An events file, a single proto, or something else? What do the entries in this table look like?